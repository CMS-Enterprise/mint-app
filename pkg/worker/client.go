package worker

import (
	"fmt"

	faktory "github.com/contribsys/faktory/client"
)

// NewClient returns a new faktory Client.
func NewClient() (*faktory.Client, error) {
	cl, err := faktory.Open()
	if err != nil {
		return nil, fmt.Errorf("error opening faktory client: %w", err)
	}
	return cl, nil
}
