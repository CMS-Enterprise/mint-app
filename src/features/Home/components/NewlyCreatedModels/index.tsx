import React, { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { Card, Grid, Icon } from '@trussworks/react-uswds';
import {
  GetModelPlansQuery,
  useGetNewlyCreatedModelPlansQuery,
  useGetNotificationSettingsQuery,
  UserNotificationPreferenceFlag,
  ViewCustomizationType
} from 'gql/generated/graphql';

import Alert from 'components/Alert';
import UswdsReactLink from 'components/LinkWrapper';
import Spinner from 'components/Spinner';
import usePagination from 'hooks/usePagination';
import { formatDateLocal } from 'utils/date';

import '../../index.scss';

const NewlyCreatedModels = () => {
  const { t: customHomeT } = useTranslation('customHome');

  const { data, loading, error } = useGetNewlyCreatedModelPlansQuery();

  const {
    data: notificationSettingsData,
    loading: notificationSettingsLoading
  } = useGetNotificationSettingsQuery();

  const models = useMemo(() => {
    return data?.modelPlanCollection || [];
  }, [data?.modelPlanCollection]);

  const { currentItems, Pagination } = usePagination<
    GetModelPlansQuery['modelPlanCollection']
  >({
    items: models,
    itemsPerPage: 3,
    loading
  });

  return (
    <Grid row gap={6}>
      <Grid desktop={{ col: 5 }} className="flex-align-self-center">
        <h2 className="margin-bottom-2">
          {customHomeT(
            `settings.${ViewCustomizationType.NEWLY_CREATED_MODEL_PLANS}.heading`
          )}
        </h2>

        <p className="line-height-body-5 font-body-md">
          {customHomeT(
            `settings.${ViewCustomizationType.NEWLY_CREATED_MODEL_PLANS}.description`
          )}
        </p>

        {!notificationSettingsLoading &&
          !notificationSettingsData?.currentUser?.notificationPreferences.newModelPlan.includes(
            UserNotificationPreferenceFlag.IN_APP
          ) && (
            <UswdsReactLink
              to="/notifications/settings"
              className="margin-top-2"
            >
              {customHomeT('getNotified')}
            </UswdsReactLink>
          )}
      </Grid>

      <Grid desktop={{ col: 7 }}>
        <>
          {error && !loading && (
            <Alert
              type="error"
              className="margin-top-4"
              heading={customHomeT('error')}
            >
              {customHomeT('errorDescription')}
            </Alert>
          )}

          {loading ? (
            <div className="display-flex flex-justify-center margin-top-4">
              <Spinner />
            </div>
          ) : (
            <>
              {currentItems.length === 0 ? (
                <Alert
                  type="info"
                  className="margin-top-4"
                  heading={customHomeT(
                    `settings.${ViewCustomizationType.NEWLY_CREATED_MODEL_PLANS}.noResultsHeading`
                  )}
                >
                  {customHomeT(
                    `settings.${ViewCustomizationType.NEWLY_CREATED_MODEL_PLANS}.noResultsDescription`
                  )}
                </Alert>
              ) : (
                <>
                  {currentItems.map(model => (
                    <Card key={model.id} className="home__card">
                      <div className="usa-collection__item border-0 padding-0 margin-0">
                        <UswdsReactLink
                          to={`/models/${model.id}/read-view`}
                          aria-label={model.modelName}
                        >
                          <span>
                            <h4 className="usa-collection__heading display-inline margin-right-1 font-body-md">
                              {model.modelName}{' '}
                              {model.abbreviation
                                ? `(${model.abbreviation})`
                                : ''}
                            </h4>
                            <Icon.ArrowForward
                              className="top-3px"
                              aria-label="forward"
                            />
                          </span>
                        </UswdsReactLink>
                      </div>{' '}
                      <div className="display-block">
                        {customHomeT('created', {
                          date: formatDateLocal(model.createdDts, 'MM/dd/yyyy')
                        })}{' '}
                        {model.modifiedDts ? (
                          <>
                            <span className="text-base-lighter margin-x-105">
                              |
                            </span>
                            {customHomeT('updated', {
                              date: formatDateLocal(
                                model.modifiedDts,
                                'MM/dd/yyyy'
                              )
                            })}
                          </>
                        ) : (
                          ''
                        )}
                        {model.discussions.length > 0 ? (
                          <>
                            <span className="text-base-lighter margin-x-105">
                              |
                            </span>
                            <Icon.Comment
                              className="text-primary margin-right-05"
                              aria-label="comment"
                            />
                            {customHomeT('discussions', {
                              count: model.discussions.length
                            })}
                          </>
                        ) : (
                          ''
                        )}
                      </div>
                    </Card>
                  ))}

                  {Pagination}
                </>
              )}
            </>
          )}
        </>
      </Grid>
    </Grid>
  );
};

export default NewlyCreatedModels;
