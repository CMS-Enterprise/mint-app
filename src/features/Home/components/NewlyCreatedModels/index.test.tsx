import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { render, screen, waitFor } from '@testing-library/react';
import {
  GetNewlyCreatedModelPlansDocument,
  GetNotificationSettingsDocument,
  GetNotificationSettingsQuery,
  GetNotificationSettingsQueryVariables,
  UserNotificationPreferenceFlag
} from 'gql/generated/graphql';
import setup from 'tests/util';

import NewlyCreatedModels from './index';

const createMockModelPlan = (
  id: string,
  name: string,
  createdDts: string,
  modifiedDts?: string | null,
  discussionsCount: number = 0
) => ({
  __typename: 'ModelPlan' as const,
  id,
  modelName: name,
  abbreviation: null,
  createdDts,
  modifiedDts,
  discussions: Array.from({ length: discussionsCount }, (_, i) => ({
    __typename: 'PlanDiscussion' as const,
    id: `discussion-${id}-${i}`
  }))
});

// Notification Settings Mocks
const mockNotificationSettingsWithInApp: MockedResponse<
  GetNotificationSettingsQuery,
  GetNotificationSettingsQueryVariables
>[] = [
  {
    request: {
      query: GetNotificationSettingsDocument
    },
    result: {
      data: {
        __typename: 'Query',
        currentUser: {
          __typename: 'CurrentUser',
          leadModelPlanCount: 0,
          notificationPreferences: {
            __typename: 'UserNotificationPreferences',
            id: 'pref-123',
            dailyDigestComplete: [],
            addedAsCollaborator: [UserNotificationPreferenceFlag.EMAIL],
            taggedInDiscussion: [],
            taggedInDiscussionReply: [],
            newDiscussionAdded: [],
            newDiscussionReply: [],
            modelPlanShared: [],
            newModelPlan: [
              UserNotificationPreferenceFlag.IN_APP,
              UserNotificationPreferenceFlag.EMAIL
            ],
            datesChanged: [],
            newDiscussionAddedNotificationType: null,
            datesChangedNotificationType: null,
            dataExchangeApproachMarkedComplete: [],
            dataExchangeApproachMarkedCompleteNotificationType: null,
            incorrectModelStatus: []
          }
        }
      }
    }
  }
];

const mockNotificationSettingsWithoutInApp: MockedResponse<
  GetNotificationSettingsQuery,
  GetNotificationSettingsQueryVariables
>[] = [
  {
    request: {
      query: GetNotificationSettingsDocument
    },
    result: {
      data: {
        __typename: 'Query',
        currentUser: {
          __typename: 'CurrentUser',
          leadModelPlanCount: 0,
          notificationPreferences: {
            __typename: 'UserNotificationPreferences',
            id: 'pref-123',
            dailyDigestComplete: [],
            addedAsCollaborator: [UserNotificationPreferenceFlag.EMAIL],
            taggedInDiscussion: [],
            taggedInDiscussionReply: [],
            newDiscussionAdded: [],
            newDiscussionReply: [],
            modelPlanShared: [],
            newModelPlan: [UserNotificationPreferenceFlag.EMAIL], // No IN_APP
            datesChanged: [],
            newDiscussionAddedNotificationType: null,
            datesChangedNotificationType: null,
            dataExchangeApproachMarkedComplete: [],
            dataExchangeApproachMarkedCompleteNotificationType: null,
            incorrectModelStatus: []
          }
        }
      }
    }
  }
];

const mockNotificationSettingsEmpty: MockedResponse<
  GetNotificationSettingsQuery,
  GetNotificationSettingsQueryVariables
>[] = [
  {
    request: {
      query: GetNotificationSettingsDocument
    },
    result: {
      data: {
        __typename: 'Query',
        currentUser: {
          __typename: 'CurrentUser',
          leadModelPlanCount: 0,
          notificationPreferences: {
            __typename: 'UserNotificationPreferences',
            id: 'pref-123',
            dailyDigestComplete: [],
            addedAsCollaborator: [],
            taggedInDiscussion: [],
            taggedInDiscussionReply: [],
            newDiscussionAdded: [],
            newDiscussionReply: [],
            modelPlanShared: [],
            newModelPlan: [], // Empty array
            datesChanged: [],
            newDiscussionAddedNotificationType: null,
            datesChangedNotificationType: null,
            dataExchangeApproachMarkedComplete: [],
            dataExchangeApproachMarkedCompleteNotificationType: null,
            incorrectModelStatus: []
          }
        }
      }
    }
  }
];

// Model Plans Mocks
const mockWithMultipleModels = [
  {
    request: {
      query: GetNewlyCreatedModelPlansDocument
    },
    result: {
      data: {
        __typename: 'Query' as const,
        modelPlanCollection: [
          createMockModelPlan(
            'model-1',
            'Innovative Care Model',
            '2024-10-01T10:00:00Z',
            '2024-10-15T14:30:00Z',
            3
          ),
          createMockModelPlan(
            'model-2',
            'Value-Based Care Initiative',
            '2024-09-15T08:00:00Z',
            null,
            0
          ),
          createMockModelPlan(
            'model-3',
            'Primary Care First',
            '2024-08-20T12:00:00Z',
            '2024-09-01T09:00:00Z',
            1
          ),
          createMockModelPlan(
            'model-4',
            'Enhanced Oncology Model',
            '2024-07-10T16:00:00Z',
            null,
            5
          ),
          createMockModelPlan(
            'model-5',
            'Medicare Advantage Innovation',
            '2024-06-05T11:00:00Z',
            '2024-06-20T15:00:00Z',
            2
          )
        ]
      }
    }
  },
  ...mockNotificationSettingsEmpty
];

const mockWithNoModels = [
  {
    request: {
      query: GetNewlyCreatedModelPlansDocument
    },
    result: {
      data: {
        __typename: 'Query' as const,
        modelPlanCollection: []
      }
    }
  },
  ...mockNotificationSettingsEmpty
];

const mockWithFewModels = [
  {
    request: {
      query: GetNewlyCreatedModelPlansDocument
    },
    result: {
      data: {
        __typename: 'Query' as const,
        modelPlanCollection: [
          createMockModelPlan(
            'model-1',
            'Test Model',
            '2024-10-01T10:00:00Z',
            '2024-10-15T14:30:00Z',
            2
          )
        ]
      }
    }
  },
  ...mockNotificationSettingsEmpty
];

describe('NewlyCreatedModels', () => {
  it('renders heading and description', async () => {
    render(
      <MemoryRouter>
        <MockedProvider mocks={mockWithNoModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText('Newly created models')).toBeInTheDocument();
    });

    expect(
      screen.getByText(
        /Check out the models created within the last six months/
      )
    ).toBeInTheDocument();
  });

  it('shows loading spinner while fetching data', () => {
    const loadingMock = [
      {
        request: {
          query: GetNewlyCreatedModelPlansDocument
        },
        result: {
          data: {
            __typename: 'Query' as const,
            modelPlanCollection: []
          }
        },
        delay: 100
      },
      ...mockNotificationSettingsEmpty
    ];

    render(
      <MemoryRouter>
        <MockedProvider mocks={loadingMock}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    expect(screen.getByTestId('spinner')).toBeInTheDocument();
  });

  it('displays alert when no models exist', async () => {
    render(
      <MemoryRouter>
        <MockedProvider mocks={mockWithNoModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText('There are no new models.')).toBeInTheDocument();
      expect(screen.getByText('Check back later.')).toBeInTheDocument();
    });
  });

  it('displays model cards after loading', async () => {
    render(
      <MemoryRouter>
        <MockedProvider mocks={mockWithMultipleModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText('Innovative Care Model')).toBeInTheDocument();
    });

    // Should display first 3 models (pagination limit is 3)
    expect(screen.getByText('Innovative Care Model')).toBeInTheDocument();
    expect(screen.getByText('Value-Based Care Initiative')).toBeInTheDocument();
    expect(screen.getByText('Primary Care First')).toBeInTheDocument();

    // Fourth and fifth models should not be visible on first page
    expect(
      screen.queryByText('Enhanced Oncology Model')
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText('Medicare Advantage Innovation')
    ).not.toBeInTheDocument();
  });

  it('displays created date for all models', async () => {
    render(
      <MemoryRouter>
        <MockedProvider mocks={mockWithFewModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText(/Created 10\/01\/2024/)).toBeInTheDocument();
    });
  });

  it('displays updated date when available', async () => {
    render(
      <MemoryRouter>
        <MockedProvider mocks={mockWithFewModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText(/Updated 10\/15\/2024/)).toBeInTheDocument();
    });
  });

  it('does not display updated date when not available', async () => {
    const mockNoUpdate = [
      {
        request: {
          query: GetNewlyCreatedModelPlansDocument
        },
        result: {
          data: {
            __typename: 'Query' as const,
            modelPlanCollection: [
              createMockModelPlan(
                'model-1',
                'Test Model',
                '2024-10-01T10:00:00Z',
                null,
                0
              )
            ]
          }
        }
      },
      ...mockNotificationSettingsEmpty
    ];

    render(
      <MemoryRouter>
        <MockedProvider mocks={mockNoUpdate}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText(/Created 10\/01\/2024/)).toBeInTheDocument();
    });

    expect(screen.queryByText(/Updated/)).not.toBeInTheDocument();
  });

  it('displays discussion count when discussions exist', async () => {
    render(
      <MemoryRouter>
        <MockedProvider mocks={mockWithFewModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText(/2 discussions/)).toBeInTheDocument();
    });
  });

  it('displays singular discussion text for one discussion', async () => {
    const mockOneDiscussion = [
      {
        request: {
          query: GetNewlyCreatedModelPlansDocument
        },
        result: {
          data: {
            __typename: 'Query' as const,
            modelPlanCollection: [
              createMockModelPlan(
                'model-1',
                'Test Model',
                '2024-10-01T10:00:00Z',
                null,
                1
              )
            ]
          }
        }
      },
      ...mockNotificationSettingsEmpty
    ];

    render(
      <MemoryRouter>
        <MockedProvider mocks={mockOneDiscussion}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText(/1 discussion$/)).toBeInTheDocument();
    });
  });

  it('does not display discussion count when no discussions exist', async () => {
    const mockNoDiscussions = [
      {
        request: {
          query: GetNewlyCreatedModelPlansDocument
        },
        result: {
          data: {
            __typename: 'Query' as const,
            modelPlanCollection: [
              createMockModelPlan(
                'model-1',
                'Test Model',
                '2024-10-01T10:00:00Z',
                null,
                0
              )
            ]
          }
        }
      },
      ...mockNotificationSettingsEmpty
    ];

    render(
      <MemoryRouter>
        <MockedProvider mocks={mockNoDiscussions}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText('Test Model')).toBeInTheDocument();
    });

    expect(screen.queryByText(/discussion/)).not.toBeInTheDocument();
  });

  it('renders links to model read view', async () => {
    render(
      <MemoryRouter>
        <MockedProvider mocks={mockWithFewModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      const link = screen.getByRole('link', { name: 'Test Model' });
      expect(link).toHaveAttribute('href', '/models/model-1/read-view');
    });
  });

  it('handles pagination correctly', async () => {
    const { user } = setup(
      <MemoryRouter>
        <MockedProvider mocks={mockWithMultipleModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    // Wait for initial data to load
    await waitFor(() => {
      expect(screen.getByText('Innovative Care Model')).toBeInTheDocument();
    });

    // First page should show first 3 models
    expect(screen.getByText('Innovative Care Model')).toBeInTheDocument();
    expect(screen.getByText('Value-Based Care Initiative')).toBeInTheDocument();
    expect(screen.getByText('Primary Care First')).toBeInTheDocument();
    expect(
      screen.queryByText('Enhanced Oncology Model')
    ).not.toBeInTheDocument();

    // Navigate to next page
    const nextButton = screen.getByRole('button', { name: /Next/i });
    await user.click(nextButton);

    // Second page should show next 2 models
    await waitFor(() => {
      expect(screen.getByText('Enhanced Oncology Model')).toBeInTheDocument();
    });
    expect(
      screen.getByText('Medicare Advantage Innovation')
    ).toBeInTheDocument();

    // First page models should not be visible
    expect(screen.queryByText('Innovative Care Model')).not.toBeInTheDocument();
  });

  it('does not display notification link when user has IN_APP enabled', async () => {
    const mocksWithInApp = [
      {
        request: {
          query: GetNewlyCreatedModelPlansDocument
        },
        result: {
          data: {
            __typename: 'Query' as const,
            modelPlanCollection: [
              createMockModelPlan(
                'model-1',
                'Test Model',
                '2024-10-01T10:00:00Z',
                '2024-10-15T14:30:00Z',
                2
              )
            ]
          }
        }
      },
      ...mockNotificationSettingsWithInApp
    ];

    render(
      <MemoryRouter>
        <MockedProvider mocks={mocksWithInApp}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText('Test Model')).toBeInTheDocument();
    });

    expect(
      screen.queryByText(/Get notified when a model is added to MINT/)
    ).not.toBeInTheDocument();
    expect(
      screen.queryByRole('link', { name: /Get notified/ })
    ).not.toBeInTheDocument();
  });

  it('displays notification link when user does not have IN_APP enabled', async () => {
    const mocksWithoutInApp = [
      {
        request: {
          query: GetNewlyCreatedModelPlansDocument
        },
        result: {
          data: {
            __typename: 'Query' as const,
            modelPlanCollection: [
              createMockModelPlan(
                'model-1',
                'Test Model',
                '2024-10-01T10:00:00Z',
                '2024-10-15T14:30:00Z',
                2
              )
            ]
          }
        }
      },
      ...mockNotificationSettingsWithoutInApp
    ];

    render(
      <MemoryRouter>
        <MockedProvider mocks={mocksWithoutInApp}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText('Test Model')).toBeInTheDocument();
    });

    await waitFor(() => {
      const link = screen.getByRole('link', {
        name: /Get notified when a model is added to MINT/
      });
      expect(link).toBeInTheDocument();
      expect(link).toHaveAttribute('href', '/notifications/settings');
      expect(link).toHaveClass('margin-top-2');
    });
  });

  it('displays notification link when newModelPlan preferences are empty', async () => {
    render(
      <MemoryRouter>
        <MockedProvider mocks={mockWithFewModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText('Test Model')).toBeInTheDocument();
    });

    await waitFor(() => {
      const link = screen.getByRole('link', {
        name: /Get notified when a model is added to MINT/
      });
      expect(link).toBeInTheDocument();
      expect(link).toHaveAttribute('href', '/notifications/settings');
    });
  });

  it('link navigates to notification settings page', async () => {
    const mocksWithoutInApp = [
      {
        request: {
          query: GetNewlyCreatedModelPlansDocument
        },
        result: {
          data: {
            __typename: 'Query' as const,
            modelPlanCollection: [
              createMockModelPlan(
                'model-1',
                'Test Model',
                '2024-10-01T10:00:00Z',
                '2024-10-15T14:30:00Z',
                2
              )
            ]
          }
        }
      },
      ...mockNotificationSettingsWithoutInApp
    ];

    render(
      <MemoryRouter>
        <MockedProvider mocks={mocksWithoutInApp}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      const link = screen.getByRole('link', {
        name: /Get notified when a model is added to MINT/
      });
      expect(link).toHaveAttribute('href', '/notifications/settings');
    });
  });

  it('matches snapshot with models', async () => {
    const { asFragment } = render(
      <MemoryRouter>
        <MockedProvider mocks={mockWithMultipleModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText('Innovative Care Model')).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with no models', async () => {
    const { asFragment } = render(
      <MemoryRouter>
        <MockedProvider mocks={mockWithNoModels}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(screen.getByText('There are no new models.')).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot while loading', () => {
    const loadingMock = [
      {
        request: {
          query: GetNewlyCreatedModelPlansDocument
        },
        result: {
          data: {
            __typename: 'Query' as const,
            modelPlanCollection: []
          }
        },
        delay: 100
      },
      ...mockNotificationSettingsEmpty
    ];

    const { asFragment } = render(
      <MemoryRouter>
        <MockedProvider mocks={loadingMock}>
          <NewlyCreatedModels />
        </MockedProvider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
