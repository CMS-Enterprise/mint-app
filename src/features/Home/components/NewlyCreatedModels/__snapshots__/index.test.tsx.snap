// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewlyCreatedModels > matches snapshot while loading 1`] = `
<DocumentFragment>
  <div
    class="grid-row grid-gap-6"
    data-testid="grid"
  >
    <div
      class="desktop:grid-col-5 flex-align-self-center"
      data-testid="grid"
    >
      <h2
        class="margin-bottom-2"
      >
        Newly created Model Plans
      </h2>
      <p
        class="line-height-body-5 font-body-md"
      >
        Check out the Model Plans created within the last six months to stay up-to-date on what’s happening at CMMI.
      </p>
    </div>
    <div
      class="desktop:grid-col-7"
      data-testid="grid"
    >
      <div
        class="display-flex flex-justify-center margin-top-4"
      >
        <span
          aria-valuetext="Loading"
          class="mint-spinner"
          data-testid="spinner"
          role="progressbar"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NewlyCreatedModels > matches snapshot with models 1`] = `
<DocumentFragment>
  <div
    class="grid-row grid-gap-6"
    data-testid="grid"
  >
    <div
      class="desktop:grid-col-5 flex-align-self-center"
      data-testid="grid"
    >
      <h2
        class="margin-bottom-2"
      >
        Newly created Model Plans
      </h2>
      <p
        class="line-height-body-5 font-body-md"
      >
        Check out the Model Plans created within the last six months to stay up-to-date on what’s happening at CMMI.
      </p>
      <a
        class="usa-link margin-top-2"
        data-discover="true"
        href="/notifications/settings"
      >
        Get notified when a Model Plan is added to MINT
      </a>
    </div>
    <div
      class="desktop:grid-col-7"
      data-testid="grid"
    >
      <li
        class="usa-card home__card"
        data-testid="Card"
      >
        <div
          class="usa-card__container"
        >
          <div
            class="usa-collection__item border-0 padding-0 margin-0"
          >
            <a
              aria-label="Innovative Care Model"
              class="usa-link"
              data-discover="true"
              href="/models/model-1/read-view"
            >
              <span>
                <h4
                  class="usa-collection__heading display-inline margin-right-1 font-body-md"
                >
                  Innovative Care Model 
                </h4>
                <svg
                  aria-label="forward"
                  class="usa-icon top-3px"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
            </a>
          </div>
           
          <div
            class="display-block"
          >
            Created 10/01/2024 
            <span
              class="text-base-lighter margin-x-105"
            >
              |
            </span>
            Updated 10/15/2024
            <span
              class="text-base-lighter margin-x-105"
            >
              |
            </span>
            <svg
              aria-label="comment"
              class="usa-icon text-primary margin-right-05"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
              />
            </svg>
            3 discussions
          </div>
        </div>
      </li>
      <li
        class="usa-card home__card"
        data-testid="Card"
      >
        <div
          class="usa-card__container"
        >
          <div
            class="usa-collection__item border-0 padding-0 margin-0"
          >
            <a
              aria-label="Value-Based Care Initiative"
              class="usa-link"
              data-discover="true"
              href="/models/model-2/read-view"
            >
              <span>
                <h4
                  class="usa-collection__heading display-inline margin-right-1 font-body-md"
                >
                  Value-Based Care Initiative 
                </h4>
                <svg
                  aria-label="forward"
                  class="usa-icon top-3px"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
            </a>
          </div>
           
          <div
            class="display-block"
          >
            Created 09/15/2024 
          </div>
        </div>
      </li>
      <li
        class="usa-card home__card"
        data-testid="Card"
      >
        <div
          class="usa-card__container"
        >
          <div
            class="usa-collection__item border-0 padding-0 margin-0"
          >
            <a
              aria-label="Primary Care First"
              class="usa-link"
              data-discover="true"
              href="/models/model-3/read-view"
            >
              <span>
                <h4
                  class="usa-collection__heading display-inline margin-right-1 font-body-md"
                >
                  Primary Care First 
                </h4>
                <svg
                  aria-label="forward"
                  class="usa-icon top-3px"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
            </a>
          </div>
           
          <div
            class="display-block"
          >
            Created 08/20/2024 
            <span
              class="text-base-lighter margin-x-105"
            >
              |
            </span>
            Updated 09/01/2024
            <span
              class="text-base-lighter margin-x-105"
            >
              |
            </span>
            <svg
              aria-label="comment"
              class="usa-icon text-primary margin-right-05"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
              />
            </svg>
            1 discussion
          </div>
        </div>
      </li>
      <div
        class=""
      >
        <nav
          aria-label="Pagination"
          class="usa-pagination mint-pagination"
        >
          <ul
            class="usa-pagination__list"
          >
            <li
              class="usa-pagination__item usa-pagination__page-no"
            >
              <button
                aria-current="page"
                aria-label="Page 1"
                class="usa-button usa-button--unstyled usa-pagination__button usa-current"
                data-testid="pagination-page-number"
                type="button"
              >
                1
              </button>
            </li>
            <li
              class="usa-pagination__item usa-pagination__page-no"
            >
              <button
                aria-label="Page 2"
                class="usa-button usa-button--unstyled usa-pagination__button"
                data-testid="pagination-page-number"
                type="button"
              >
                2
              </button>
            </li>
            <li
              class="usa-pagination__item usa-pagination__arrow"
            >
              <button
                aria-label="Next page"
                class="usa-button usa-button--unstyled usa-pagination__link usa-pagination__next-page"
                data-testid="pagination-next"
                type="button"
              >
                <span
                  class="usa-pagination__link-text"
                >
                  Next
                </span>
                <svg
                  aria-hidden="true"
                  class="usa-icon"
                  focusable="false"
                  height="1em"
                  role="img"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NewlyCreatedModels > matches snapshot with no models 1`] = `
<DocumentFragment>
  <div
    class="grid-row grid-gap-6"
    data-testid="grid"
  >
    <div
      class="desktop:grid-col-5 flex-align-self-center"
      data-testid="grid"
    >
      <h2
        class="margin-bottom-2"
      >
        Newly created Model Plans
      </h2>
      <p
        class="line-height-body-5 font-body-md"
      >
        Check out the Model Plans created within the last six months to stay up-to-date on what’s happening at CMMI.
      </p>
      <a
        class="usa-link margin-top-2"
        data-discover="true"
        href="/notifications/settings"
      >
        Get notified when a Model Plan is added to MINT
      </a>
    </div>
    <div
      class="desktop:grid-col-7"
      data-testid="grid"
    >
      <div
        class="usa-alert usa-alert--info flex margin-top-4"
        data-testid="alert"
      >
        <div
          class="usa-alert__body"
        >
          <h4
            class="usa-alert__heading"
          >
            There are no new Model Plans.
          </h4>
          <p
            class="usa-alert__text"
          >
            <span>
              Check back later.
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
